app/src/lib/is-git-on-path.ts

```diff
@@ -29,9 +28,20 @@ export function isGitOnPath(): Promise<boolean> {
         resolve(code === 0)
       })
       return
-    }
+    })
+  }
+
+  if (__LINUX__) {
+    return new Promise<boolean>((resolve, reject) => {
+      const process = spawn('which', ['git'])
+
+      // `which` will return 0 when the executable
+      // is found under PATH, or 1 if it cannot be found
+      process.on('close', function(code) {
+        resolve(code === 0)
+      })
+    })
+  }
 
-    // in case you're on a non-Windows/non-macOS platform
-    resolve(false)
-  })
+  return Promise.resolve(false)
```

Comment:
I'm not a huge fan of that style of that control flow even though that's totally my previous work - I actually broke all Linux users with an unhandled rejected promise ages ago in #2807 and didn't notice it because I wasn't running up the app.

I'm going to leave the `Promise.resolve(false)` here as that mirrors the current behaviour, but if someone wants to follow up with testing this out again I'm happy to revisit this.

Reply:
That's cool, was just thinking about consistency. If this is what you prefer over the jumble in `shared.ts` would it be worth opening a `tech-debt` issue to sort out that file?

Reply:
@JordanMussi Yeah, I'll follow up after this lands.